namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -0,0 +1,105 @@
+package io.buoyant.namer.dnssrv
+
+import java.io.IOException
+import java.net.InetSocketAddress
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.TimeUnit
+
+import com.twitter.finagle._
+import com.twitter.finagle.stats.{Stat, StatsReceiver}
+import com.twitter.logging.Logger
+import com.twitter.util.Activity.State
+import com.twitter.util._
+import org.xbill.DNS
+
+class DnsSrvNamer(prefix: Path, resolver: DNS.Resolver, refreshInterval: Duration, stats: StatsReceiver, pool: FuturePool)(implicit val timer: Timer)
+  extends Namer {
+
+  override def lookup(path: Path): Activity[NameTree[Name]] = memoizedLookup(path)
+
+  private val success = stats.counter("lookup_successes_total")
+  private val failure = stats.counter("lookup_failures_total")
+  private val zeroResults = stats.counter("lookup_zero_results_total")
+  private val latency = stats.stat("request_duration_seconds")
```

Comment:
as mentioned below, this should probably be measured in millis (and have suffix `_ms` like other finagle metrics)